CommandLine/emake/Makefile

```diff
@@ -1,35 +1,46 @@
-CXX=g++
-CXXFLAGS=-g -std=c++11 -c -Wall -I. -I../../CompilerSource
+BINARY := ../../emake
+SRC_DIR := .
+OBJ_DIR := .eobjs
 
 ifeq ($(OS),Windows_NT)
-	LDFLAGS=-lboost_system-mt -Wl,--no-as-needed -Wl,-rpath=./ -lboost_program_options-mt -lboost_iostreams-mt -lboost_filesystem-mt -lboost_system-mt -lpthread
+	OS_LIBS=-lboost_system-mt -Wl,--no-as-needed -Wl,-rpath=./ -lboost_program_options-mt -lboost_iostreams-mt -lboost_filesystem-mt -lboost_system-mt -lpthread
 else
-	LDFLAGS=-lboost_system -Wl,--no-as-needed -Wl,-rpath=./ -lboost_program_options -lboost_iostreams -lboost_filesystem -lboost_system -lpthread -ldl
+	OS_LIBS=-lboost_system -Wl,--no-as-needed -Wl,-rpath=./ -lboost_program_options -lboost_iostreams -lboost_filesystem -lboost_system -lpthread -ldl
 endif
-	
-SOURCES=$(shell find . -name "*.cpp")
-OBJECTS=$(addprefix .eobjs/,$(SOURCES:.cpp=.o))
-EXECUTABLE=../../emake
```

Comment:
perhaphs we should replace ../../ with and ENIGMA_ROOT var incase anyone ever tries to move this